src/main/java/com/ldtteam/blockout/views/View.java

```diff
@@ -94,12 +100,12 @@ public void drawSelfLast(final MatrixStack ms, final int mx, final int my)
         ms.translate(paddedX, paddedY, 0.0d);
 
         // Translate Mouse into the View
-        final int drawX = mx - paddedX;
-        final int drawY = my - paddedY;
+        final double drawX = mx - paddedX;
+        final double drawY = my - paddedY;
 
         for (final Pane child : children)
         {
-            if (child.isVisible())
+            if (childIsVisible(child) && child.isVisible())
```

Comment:
A very small thing, but I had to look twice to notice that they were not the exact same condition. It might be helpful to rename the method to ``childWithinBounds`` or something?

Reply:
this mechanism probably needs a visit in the future, so can be renamed later, gonna create issue